spec/spec.md

```diff
@@ -451,14 +451,32 @@ This profile currently assumes that ID Token and a single VP passed as a VP Toke
 
-#### ID Token Validation
+#### [[def:Validation of Authorization Response]]
 
-ID Token validation rules defined in section 10 of [[ref: SIOPv2 ID1]] MUST be followed. The ID Token MUST be signed by the End-User's DID.
+The following checks MUST be made by the Verifier upon receiving the Authorization Response. Verifiers MAY perform any additional checks according to their trust framework/policies.
 
-- `iss` claim MUST be `https://self-issued.me/v2/openid-vc`.
-- Signature on the ID Token MUST be validated.
-  - Validation is performed against the key obtained from a DID Document. DID Document MUST be obtained by resolving a Decentralized Identifier included in the `sub` claim using DID Resolution.
-- `sub` claim MUST equal the `id` property in the DID Document.
+##### ID Token Validation
+
+To validate the received ID Token, the Verifier MUST do the following:
+
+1. The Verifier MUST validate ID Token according to the rules defined in section 12.1 of [[ref: SIOPv2 ID1]].
+2. The ID Token MUST be signed by the End-User's DID.
+3. `iss` claim MUST be `https://self-issued.me/v2/openid-vc`.
+4. The signature on the ID Token MUST be validated. Validation is performed against the key obtained from a DID Document. DID Document MUST be obtained by resolving a Decentralized Identifier included in the `sub` claim using DID Resolution. If a DID Doc contains multiple keys, kid in the header is used to identify which key to use.
+5. The DID in the `kid` and `sub` claim MUST match.
+6. `sub` claim MUST equal the `id` property in the DID Document.
+7. The `_vp_token` claim MUST be present and contain a presentation_submission with a valid descriptor map.
+
+##### VP Token Validation
+
+To Validate the VP Token received, the Verifier MUST do the following:
+
+1. The signature on the VP Token MUST be validated. Validation is performed against the key obtained from a DID Document. DID Document MUST be obtained by resolving a Decentralized Identifier included in the `sub` claim using DID Resolution. If a DID Doc contains multiple keys, kid in the header is used to identify which key to use.
+2. The DID in the `kid` and `sub` claim MUST match.
+3. Using the descriptor map obtained from the ID Token, the Verifier MUST determine the number of VPs returned in the VP Token and identify the in which VP requested VC(s) are included.
+4. Using the presentation definition from the Authorization Request, the Verifier must confirm that the VC meets all requested criteria using the mechanisms outlined in Section 4.3 of [[ref: Presentation Exchange v1.0.0]].
+5. The signature on the VCs embedded in the VPs must be validated.
+6. It is highly recommened that the Verifier permforms Linked Domain Verification of the Issuer's DID as defined in [[ref: Linked Domain Verification]].
```

Comment:
```suggestion
Verifiers MUST go through (at least) the following steps before trusting/using any of the contents of a VP Token:

1. Validate the signature on the VP Token. Validation is performed against the key obtained from a DID Document. DID Document MUST be obtained by resolving a Decentralized Identifier included in the `sub` claim using DID Resolution. If a DID Doc contains multiple keys, kid in the header is used to identify which key to use.
2. Check that the DID value in the `kid` and `sub` claims match.
3. Determine the number of VPs returned in the VP Token and identify in which VP requested VC(s) are included, using the descriptor map obtained from the ID Token.
4. Confirm that the VC meets all requested criteria using the mechanisms outlined in Section 4.3 of [[ref: Presentation Exchange v1.0.0]], using the presentation definition from the Authorization Request.
5. Validate signature(s) on each VC(s).
6. It is highly recommened that the Verifier permforms Linked Domain Verification of the Issuer's DID as defined in [[ref: Linked Domain Verification]].
```

same comment as above - step 2 feels like part of 1.
also there is no such thing as a signature on a VP Token.. verifier needs to take out VPs from a VP Token to validate signature(s) on the VP(s)